lib/include/openamp/virtio.h

```diff
@@ -497,6 +509,38 @@ static inline int virtio_reset_device(struct virtio_device *vdev)
 	return 0;
 }
 
+/**
+ * @brief Allocate buffer from the virtio device
+ *
+ * @param vdev	Pointer to virtio device structure.
+ * @param size	Allocated buffer size.
+ * @param align	Allocated buffer alignment.
+ *
+ * @return The allocated buffer address.
+ */
+static inline void *virtio_alloc_buf(struct virtio_device *vdev,
+				     size_t size, size_t align)
+{
+	if (!vdev->mmops || !vdev->mmops->alloc)
+		return NULL;
```

Comment:
The change mostly looks good.
Only concern is about return value. How to differentiate if op is not registered or buffer allocation actually failed?

Also is it fatal if buffer allocation is failed ?? Do we need assertion ?

Reply:
Done

Reply:
Add return value for `virtio_alloc/free_buff()`